src/main/java/com/google/finapp/FinAppService.java

```diff
@@ -102,6 +103,30 @@ public void createCustomerRole(
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void moveAccountBalance(
+      MoveAccountBalanceRequest request,
+      StreamObserver<MoveAccountBalanceResponse> responseObserver) {
+    Map<ByteArray, BigDecimal> accountBalances;
+    ByteArray fromAccountId = ByteArray.copyFrom(request.getFromAccountId().toByteArray());
+    ByteArray toAccountId = ByteArray.copyFrom(request.getToAccountId().toByteArray());
+    try {
+      accountBalances =
+          spannerDao.moveAccountBalance(
+              fromAccountId, toAccountId, new BigDecimal(request.getAmount()));
+    } catch (SpannerDaoException e) {
+      responseObserver.onError(Status.fromThrowable(e).asException());
```

Comment:
If the user gives us a negative or too-high transfer amount then `moveAccountBalance` throws an `IllegalArgumentException`.  Since that isn't caught here we get the same not-nice error as in @voulg's comment above:

```
Rpc failed with status code 2, error message:
```

Ideally we'd set the right gRPC error code ([3: `INVALID_ARGUMENT` ](https://grpc.github.io/grpc/core/md_doc_statuscodes.html)) and a nice user-facing error message here.

Example failing calls that cause the same error:

```
grpc_cli call ... MoveAccountBalance 'from_account_id: "..." to_account_id: "..." amount: "50000"'
```
```
grpc_cli call ... MoveAccountBalance 'from_account_id: "..." to_account_id: "..." amount: "-500"'
```
```
grpc_cli call ... MoveAccountBalance 'from_account_id: "..." to_account_id: "..." amount: "nop"'
```